### PR TITLE
Implement syntax highlighting.

### PR DESCRIPTION
**Pull request by fabianschwarzfritz**
- Uses highlight js library
 - highlights block by block
 - highlights in expand as well in not expanded view
 - Uses the same syntax highlighter as github does

Detect highlight file type by file ending.
 - This manual detection mode is better than the automatic mode as we are highlighting per block (=per line)

Rockstar challenge :) ; :guitar: https://www.youtube.com/watch?v=0WzG64syKHA

Fixes #449

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="http://app.review.ninja/reviewninja/review.ninja/pull/838" target="_blank">ReviewNinja

File: src/client/directives/templates/diff.html [hunk 1]
```diff
@@ -16,7 +16,7 @@
                     <i class="fa fa-info fa-stack-1x text-primary"></i>
                 </span>
             </a>
-            <span ng-click="expanded=!expanded; $event.stopPropagation();">
```
Comment (gruiz17) [inline src/client/directives/templates/diff.html]: We shouldn't be using $event.stopPropagation here. !fix

Comment (gruiz17) [inline src/client/directives/templates/diff.html]: !fixed

File: src/client/directives/templates/diff.html [hunk 2]
```diff
@@ -34,13 +34,13 @@
             <table class="margin">
                 <tr ng-repeat="line in (expanded ? file.file : file.patch) track by $index">
                     <td class="select">
-                        <span class="octicon octicon-issue-opened text-warning" 
+                        <span class="octicon octicon-issue-opened text-warning"
```
Comment (gruiz17) [inline src/client/directives/templates/diff.html]: we shouldn't use refStarts here !fix

Comment (gruiz17) [inline src/client/directives/templates/diff.html]: !fixed

**Pull request closed without merging**